lib/Dialect/FIRRTL/Transforms/ExpandWhens.cpp

```diff
@@ -578,13 +702,28 @@ void ModuleVisitor::visitStmt(WhenOp whenOp) {
   processWhenOp(whenOp, /*outerCondition=*/{});
 }
 
+bool ModuleVisitor::usesSubwordUninit(Operation *op) {
+  if (!op || subwordUninit.size() == 0)
+    return false;
+  if (auto b = subwordInvalidMemo.lookup(op))
+    return b;
+  for (auto val : op->getOperands()) {
+    if (subwordUninit.contains(val) || usesSubwordUninit(val.getDefiningOp())) {
+      subwordInvalidMemo[op] = true;
+      return true;
+    }
+  }
+  subwordInvalidMemo[op] = false;
+  return false;
+}
```

Comment:
- We don't allow recursive functions on arbitrary IR, it has a bad habit of overflowing the stack. You can store operand iterators to a SmallVector.
- You don't really need to cache the positive (uses bad value) state.  So you could switch this to a set `hasBeenChecked` or something?  This doesn't really matter, but not storing a "value" should reduce the memory required for the cache

Reply:
I've changed the check to an iterative solution. Let me know if you'd like further changes to that implementation. Currently the vector is passed to the checker to avoid frequent reallocation.